src/osd/OSD.cc

```diff
@@ -6833,11 +6833,16 @@ void OSD::sched_scrub()
         break;
       }
 
+      if ((scrub.deadline >= now) && !(time_permit && load_is_low)) {
+        dout(10) << __func__ << " not scheduling scrub of " << scrub.pgid << " due to "
+                 << (!time_permit ? "time not permit" : "load not low") << dendl;
+        break;
```

Comment:
The logic after your modification is changed: in original code, if ((scrub.deadline >= now) && !(time_permit && load_is_low)) is false, this func will continue, and check next_scrub_stamp, but your in your change, will break "while" ...

This deadline is dedicated to each scrubjob, so here should be continue, not break...